specs/altair/beacon-chain.md

```diff
@@ -311,10 +320,16 @@ def get_base_reward_per_increment(state: BeaconState) -> Gwei:
 
 #### `get_base_reward`
 
-*Note*: The function `get_base_reward` is modified with the removal of `BASE_REWARDS_PER_EPOCH`.
+*Note*: The function `get_base_reward` is modified with the removal of `BASE_REWARDS_PER_EPOCH` and the use of increment based accounting.
 
 ```python
 def get_base_reward(state: BeaconState, index: ValidatorIndex) -> Gwei:
+    """
+    Return the base reward for the validator defined by ``index`` with respect to the current ``state``.
+
+    Note: A validator can optimally earn one base reward per epoch over a long time horizon.
```

Comment:
We should make a note of the difference with phase0. It's still the same amount of ETH, but the base reward naming changed from 4 (`BASE_REWARDS_PER_EPOCH`) to just 1 big one per epoch + weight logic.

Reply:
Also, the precision changed:

Phase0:
```
effective_balance * BASE_REWARD_FACTOR // integer_squareroot(total_balance) // BASE_REWARDS_PER_EPOCH
```

Altair:
```
EFFECTIVE_BALANCE_INCREMENT * BASE_REWARD_FACTOR // integer_squareroot(get_total_active_balance(state))
```

Napkin math: assuming an effective balance of ~32, that's 5 bits of extra precision that don't get rounded down. Then it divides by `BASE_REWARDS_PER_EPOCH = 4`, 2 bits less, but it repeats 4 times, so the effect may not be as big there.

The higher the precision, the less the base reward gets "rounded down". So those ~5 bits = 32 gwei (pessimistic maximum) lost per validator per attestation when forking to Altair. In the grand scheme of things it's a tiny rounding thing, but it's about validator rewards, so I think we want to get it right.

Reply:
I'll make an issue to review the precision of Altair rewards, but I don't think it's significant enough to block this PR, especially since rewards change in other bigger ways.